crates/ruff_python_formatter/src/statement/suite.rs

```diff
@@ -116,6 +117,38 @@ impl FormatRule<Suite, PyFormatContext<'_>> for FormatSuite {
         for statement in iter {
             if is_class_or_function_definition(last) || is_class_or_function_definition(statement) {
                 match self.kind {
+                    SuiteKind::TopLevel if source_type.is_stub() => match (last, statement) {
+                        // Check if the statements are class definitions of the
+                        // same type, containing only an ellipsis.
+                        (
+                            Stmt::ClassDef(ast::StmtClassDef {
+                                arguments: Some(last_args),
+                                body: last_body,
+                                ..
+                            }),
+                            Stmt::ClassDef(ast::StmtClassDef {
+                                arguments: Some(args),
+                                body,
+                                ..
+                            }),
+                        ) if last_args.len() == args.len()
+                            && last_args.args.iter().zip(args.args.iter()).all(
+                                |(last_arg, arg)| match (last_arg, arg) {
+                                    (Expr::Name(last_name), Expr::Name(name)) => {
+                                        last_name.id == name.id
+                                    }
+                                    _ => false,
+                                },
+                            )
+                            && contains_only_an_ellipsis(last_body)
+                            && contains_only_an_ellipsis(body) =>
```

Comment:
Nit: It could make sense to move this check above the `args` check because it is relatively cheap (compared with zipping and comparing all arguments)